File: SofiForce.Models/Models/SearchModels/ItemSearchModel.cs
using System;
using System.Collections.Generic;

namespace Models
{
    public class ItemSearchModel : BaseSearchModel
    {
        public Int32? ItemId { get; set; }

        public Int32? VendorId { get; set; }

        public string ItemCode { get; set; }
        public string ItemName { get; set; }

        public int? IsNewAdded { get; set; }
        public int? IsNewStocked { get; set; }
        public int? HasPromotion { get; set; }
        public int? IsActive { get; set; }

        public decimal? PublicPrice { get; set; }
        public decimal? ClientPrice { get; set; }

        public Int32? StoreId { get; set; }


    }
}

File: SofiForce.Web/ClientApp/src/app/core/Models/SearchModels/ItemSearchModel.ts
import { BaseSearchModel } from "./baseSearchModel";

export interface ItemSearchModel extends BaseSearchModel {
    itemId: number | null;
    storeId: number | null;
    storeCode: string;
    vendorId: number | null;
    vendorCode:string;
    itemCode: string;
    itemName: string;
    isNewAdded: number;
    isNewStocked: number;
    hasPromotion: number;
    isActive: number;
    
    publicPrice:number;
    clientPrice:number;
}